algorithms/kernel/dtrees/dtrees_model.cpp

```diff
@@ -204,7 +207,35 @@ void setNode(DecisionTreeNode& node, int featureIndex, double response)
     node.featureValueOrResponse = response;
 }
 
-services::Status addSplitNodeInternal(data_management::DataCollectionPtr& serializationData,size_t treeId, size_t parentId, size_t position, size_t featureIndex, double featureValue, size_t& res)
+void setProbabilities(const size_t treeId, const size_t nodeId, const size_t response,
```

Comment:
why this function doesn`t return Status?

Reply:
All corner cases is processed outside this function